### ISSUE TITLE
Automatic suggestion of reference tags

### ISSUE THREAD

**Issue by Sadrettin86** [OpenRefine/OpenRefine#6814]
I want the tags in the References to be suggested automatically. Every time, I manually type the tags url, title, retrieved, language of work, archive url, archive date, publisher, author, etc. After I type them, I check if I have written the correct property number. Whereas, for example, OpenRefine could have suggested P854 (Reference URL) to me **before I started typing**. (P854: https://w.wiki/9YPi)

A form that comes with ready-made reference tags would be useful.
Thank you. Best regards.

![image](https://github.com/user-attachments/assets/480897d9-3034-4b6a-9005-256abb389a87)

**Comment by wetneb** [comment]
In Wikidata, when creating a new reference, one indeed gets suggestions before starting to type:
![image](https://github.com/user-attachments/assets/62f46edf-65ac-4bf0-bf01-45c62475f1df)

The suggested properties are indeed relevant for references.
This sort of auto-completion is likely available from the API and could be replicated in OpenRefine.

**Comment by Sadrettin86** [comment]
For example, in Wikipedia, certain source tags are in the form of a form. It is easier to fill out this form. Of course, it cannot be the same as in Wikipedia. However, when I want to add a source, it would be easy to fill out the form that includes commonly used tags such as url, title, access date, archive url and archive access date.

![image](https://github.com/user-attachments/assets/99313230-7b48-41e3-9821-821b3a36b3f3)

**Comment by wetneb** [comment]
Marking this issue as a good first issue, with the goal of replicating wikidata's behaviour (without any custom extension) inside OpenRefine's schema editor.

**Comment by wetneb** [comment]
This issue is about the Wikibase integration in OpenRefine, so it's useful to be familiar with those features before tackling this issue.

To do so, you could follow [this tutorial](https://www.wikidata.org/wiki/Wikidata:Tools/OpenRefine/Editing/Tutorials/Basic_editing), which will lead you to creating a Wikibase schema, add a statement in it and add a reference to it. Specifically, the addition of a reference is done in [Section 4: Detecting and fixing issues](https://www.wikidata.org/wiki/Wikidata:Tools/OpenRefine/Editing/Tutorials/Basic_editing#Detecting_and_fixing_issues).

**Comment by nadaa** [comment]
Hi,
I want to work on this.
Thanks

**Comment by estebanpuyanas** [comment]
Hi @wetneb, I'm interested in contributing to OpenRefine. Any chance I can work on this issue?

**Comment by hariprabhu571** [comment]
Hi @wetneb, I'm interested, could you assign me ?

**Comment by SoryRawyer** [comment]
Hi @hariprabhu571, as I mentioned on another issue, please only comment on one issue at a time when looking to be assigned. I've gone ahead and assigned you to another issue. Thanks for your interest in contributing!

**Comment by ElishevaShmuel** [comment]
Hi @SoryRawyer @wetneb , I would like to contribute to the open-source code by solving this task. Could you please assign it to me?

**Comment by Srishtig0711** [comment]
Hi @SoryRawyer @wetneb ! I would like to work on this issue if it is still available. Please let me know if I can proceed. And if this issue is taken, can you please assign me another issue similar to this!

**Comment by SoryRawyer** [comment]
Ah sorry for missing this @ElishevaShmuel. Are you still interested in the task? If not, I'll assign @Srishtig0711. @Srishtig0711 in the meantime, you're more than welcome to look through other issues and comment on one that looks appealing to you.

**Comment by ElishevaShmuel** [comment]
Actualy, I’m currently focused on something else. Thank you

**Comment by SoryRawyer** [comment]
Thanks for the response, and my apologies again for missing your message. @Srishtig0711 I'll go ahead and assign this to you. Please feel free to reach out if you have any questions!

**Comment by rawmeat21** [comment]
Seems like this issue is still open. Is anyone working on this currently? If not, I'd like to work on this.

**Comment by durkagiri21-ship-it** [comment]
Hi, I'm interested in working on this issue as one of my first contributions to OpenRefine. Could you provide any pointers on where the reference property suggestion logic currently lives in the codebase? Happy to start investigating if assigned.

**Comment by magdmartin** [comment]
I assigned the issue to @rawmeat21  on a first-come, first-served basis.  I invite you to review our [Your first code pull request](https://openrefine.org/docs/technical-reference/code-contributions ) and related documentation to get started.
